test/__tests__/bopt_test.ml

```diff
@@ -0,0 +1,263 @@
+  test "wrap2 - raise, handler = true" (fun t ->
+    let handler = function | Invalid_argument _ -> true | _ -> false in
+    deepEqual t
+      (wrap2 ~handler (fun _ _ -> raise (Invalid_argument "")) 2 7) (None));
+  test "wrap2 - raise, handler = false" (fun t ->
+    let handler = function | Invalid_argument _ -> false | _ -> true in
+    try 
+      ignore @@ wrap2 ~handler (fun _ _ -> raise (Invalid_argument "")) 2 7;
+      fail t
+    with
+      | Invalid_argument _ -> pass t);
+  test "wrap2 - no raise" (fun t ->
+    deepEqual t
+      (wrap2 (+) 2 7) (Some 9));
+
+  test "isSome - Some _" (fun t ->
+    deepEqual t (isSome (Some 2)) true);
+  test "isSome - None" (fun t ->
+    deepEqual t (isSome None) false);
+
+  test "isNone - Some _" (fun t ->
+    deepEqual t (isNone (Some 2)) false);
+  test "isNone - None" (fun t ->
+    deepEqual t (isNone None) true);
+
+  test "equal - =, Some _ = Some _" (fun t ->
+    deepEqual t (equal (=) (Some 2) (Some 2)) true);
+  test "equal - <>, Some _ <> Some _" (fun t ->
+    deepEqual t (equal (<>) (Some 2) (Some 2)) false);
+  test "equal - <>, Some _ <> None" (fun t ->
+    deepEqual t (equal (<>) (Some 2) None) false);
+  test "equal - <>, None <> Some _" (fun t ->
+    deepEqual t (equal (<>) None (Some 2)) false);
+  test "equal - <>, None = None" (fun t ->
+    deepEqual t (equal (<>) None None) true);
+
+  test "compare - Some _ = Some _" (fun t ->
+    deepEqual t (compare Comparison.compare (Some 2) (Some 2)) Equal);
+  test "compare - Some _ < Some _" (fun t ->
+    deepEqual t (compare Comparison.compare (Some 2) (Some 3)) Less);
+  test "compare - Some _ > None" (fun t ->
+    deepEqual t (compare Comparison.compare (Some 2) None) Greater);
+  test "comapre - None < Some _" (fun t ->
+    deepEqual t (compare Comparison.compare None (Some 2)) Less);
+  test "compare - None = None" (fun t ->
+    deepEqual t (compare Comparison.compare None None) Equal);
+
+  test "get - Some _" (fun t ->
+    deepEqual t (get "b" (Some "a")) "a");
+  test "get - None" (fun t ->
+    deepEqual t (get "b" None) "b");
+  
+  test "getOr - Some _" (fun t ->
+    deepEqual t (getOr ~default:"b" (Some "a")) "a");
+  test "getOr - None" (fun t ->
+    deepEqual t (getOr ~default:"b" None) "b");
+
+  test "getOrRaise - Some _" (fun t ->
+    deepEqual t (getOrRaise (Some "a")) "a");
+  test "getOrRaise - None" (fun t ->
+    try
+      ignore @@ getOrRaise None;
+      fail t
+    with
+      | _ -> pass t);
+
+  test "getLazy - Some _" (fun t ->
+    deepEqual t (getLazy  (fun () -> "b") (Some "a")) "a");
+  test "getLazy - None" (fun t ->
+    deepEqual t (getLazy (fun () -> "b") None) "b");
+
+  test "forEach - Some _" (fun t ->
+    plan t 1;
+    forEach (fun x -> deepEqual t x "a") (Some "a"));
+  test "forEach - None" (fun t ->
+    forEach (fun _ -> fail t) None);
+
+  test "map - Some _" (fun t ->
+    deepEqual t (map (( *) 3) (Some 7)) (Some 21));
```

Comment:
Also we are getting this warning, do you know why? `Warning 2: this is not the end of a comment.`